porch/engine/pkg/engine/error.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package engine
+
+import "fmt"
+
+type UnsupportedFunctionError struct {
```

Comment:
should this be part of the FunctionRuntime contract?
(if so we'd want to declare this error in kpt: https://github.com/GoogleContainerTools/kpt/tree/main/pkg/fn)

please check with @droot if he'd be ok with moving this there

Reply:
Moving under `pkg/fn` sounds good to me.

Reply:
I cherry-picked the commit from Justin's PR.